File: CS/WindowsApplication34/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;
using DevExpress.Data.PivotGrid;
using DevExpress.Utils.Menu;
using DevExpress.XtraPivotGrid.ViewInfo;
using System.Reflection;

namespace WindowsApplication34
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            pivotGridControl1.DataSource = CreateTable(20);
            pivotGridControl1.PopupMenuShowing +=new PopupMenuShowingEventHandler(pivotGridControl1_PopupMenuShowing); 

            pivotGridControl1.Fields.AddDataSourceColumn("Type", DevExpress.XtraPivotGrid.PivotArea.RowArea);
            pivotGridControl1.Fields.AddDataSourceColumn("Product", DevExpress.XtraPivotGrid.PivotArea.RowArea);
            PivotGridField fieldYear = new PivotGridField("", PivotArea.ColumnArea);
            fieldYear.DataBinding = new DataSourceColumnBinding("Date", PivotGroupInterval.DateYear);
            fieldYear.Name = "FieldYear";
            fieldYear.Caption = fieldYear.Name;
            pivotGridControl1.Fields.AddRange(new PivotGridField[] { fieldYear });

            PivotGridField dataField = pivotGridControl1.Fields.AddDataSourceColumn("Number", DevExpress.XtraPivotGrid.PivotArea.DataArea);
            dataField.Options.AllowRunTimeSummaryChange = true;

        }

        List<string> necessaryItems = new List<string>(new string[] { "Sum", "Max", "Average", "Min", "Count" });
        void pivotGridControl1_PopupMenuShowing(object sender, PopupMenuShowingEventArgs e)
        {         
            if (e.MenuType == PivotGridMenuType.HeaderSummaries)
            {
                int i = 0;
                while (i < e.Menu.Items.Count)
                {
                    if (!necessaryItems.Contains(e.Menu.Items[i].Caption))
                        e.Menu.Items.RemoveAt(i);
                    else
                        i++;                    
                }
            }                
        }

        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Type", typeof(string));
            tbl.Columns.Add("Product", typeof(string));
            tbl.Columns.Add("Date", typeof(DateTime));
            tbl.Columns.Add("Number", typeof(int));

            Random r = new Random();
            for (int i = 0; i < RowCount; i++)

                for (int j = 0; j < RowCount; j++)
                    tbl.Rows.Add(new object[] { String.Format("Type {0}", i % 2), String.Format("Product {0}", i % 3), DateTime.Now.AddYears(j % 3), r.Next(2) });
            return tbl;
        }

        private void pivotGridControl1_MouseMove(object sender, MouseEventArgs e)
        {
            PivotGridControl pivot = (PivotGridControl)sender;
            PivotGridHitInfo hitInfo = pivot.CalcHitInfo(e.Location);
            if (hitInfo.HitTest != PivotGridHitTest.HeadersArea) return;
            if (hitInfo.HeaderField != null && hitInfo.HeaderField.Area == PivotArea.DataArea && hitInfo.HeaderField.Options.AllowRunTimeSummaryChange)
            {
                PropertyInfo pInfo = typeof(PivotGridHeadersAreaHitInfo).GetProperty("HeadersViewInfo", BindingFlags.Instance | BindingFlags.NonPublic);
                PivotHeadersViewInfoBase headersViewInfo = pInfo.GetValue(hitInfo.HeadersAreaInfo, null) as PivotHeadersViewInfoBase;
                PivotHeaderViewInfo headerViewInfo = headersViewInfo[ GetFieldItem( pivot, hitInfo.HeaderField ) ] as PivotHeaderViewInfo;

                MethodInfo mInfo = typeof(PivotHeaderViewInfo).GetMethod("ShowSummariesMenu", BindingFlags.NonPublic | BindingFlags.Instance );
                mInfo.Invoke(headerViewInfo, new object[] { });
            }            
        }

        private DevExpress.XtraPivotGrid.Data.PivotFieldItemBase GetFieldItem(PivotGridControl pivot, PivotGridField field)
        {
            return ((IPivotGridViewInfoDataOwner)pivot).DataViewInfo.GetFieldItem(field);
        }

        

    }
}

File: VB/WindowsApplication34/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraPivotGrid.ViewInfo
Imports System.Reflection

Namespace WindowsApplication34

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            pivotGridControl1.DataSource = CreateTable(20)
            AddHandler pivotGridControl1.PopupMenuShowing, New PopupMenuShowingEventHandler(AddressOf pivotGridControl1_PopupMenuShowing)
            pivotGridControl1.Fields.AddDataSourceColumn("Type", PivotArea.RowArea)
            pivotGridControl1.Fields.AddDataSourceColumn("Product", PivotArea.RowArea)
            Dim fieldYear As PivotGridField = New PivotGridField("", PivotArea.ColumnArea)
            fieldYear.DataBinding = New DataSourceColumnBinding("Date", PivotGroupInterval.DateYear)
            fieldYear.Name = "FieldYear"
            fieldYear.Caption = fieldYear.Name
            pivotGridControl1.Fields.AddRange(New PivotGridField() {fieldYear})
            Dim dataField As PivotGridField = pivotGridControl1.Fields.AddDataSourceColumn("Number", PivotArea.DataArea)
            dataField.Options.AllowRunTimeSummaryChange = True
        End Sub

        Private necessaryItems As List(Of String) = New List(Of String)(New String() {"Sum", "Max", "Average", "Min", "Count"})

        Private Sub pivotGridControl1_PopupMenuShowing(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs)
            If e.MenuType = PivotGridMenuType.HeaderSummaries Then
                Dim i As Integer = 0
                While i < e.Menu.Items.Count
                    If Not necessaryItems.Contains(e.Menu.Items(i).Caption) Then
                        e.Menu.Items.RemoveAt(i)
                    Else
                        i += 1
                    End If
                End While
            End If
        End Sub

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Type", GetType(String))
            tbl.Columns.Add("Product", GetType(String))
            tbl.Columns.Add("Date", GetType(Date))
            tbl.Columns.Add("Number", GetType(Integer))
            Dim r As Random = New Random()
            For i As Integer = 0 To RowCount - 1
                For j As Integer = 0 To RowCount - 1
                    tbl.Rows.Add(New Object() {String.Format("Type {0}", i Mod 2), String.Format("Product {0}", i Mod 3), Date.Now.AddYears(j Mod 3), r.Next(2)})
                Next
            Next

            Return tbl
        End Function

        Private Sub pivotGridControl1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim pivot As PivotGridControl = CType(sender, PivotGridControl)
            Dim hitInfo As PivotGridHitInfo = pivot.CalcHitInfo(e.Location)
            If hitInfo.HitTest <> PivotGridHitTest.HeadersArea Then Return
            If hitInfo.HeaderField IsNot Nothing AndAlso hitInfo.HeaderField.Area = PivotArea.DataArea AndAlso hitInfo.HeaderField.Options.AllowRunTimeSummaryChange Then
                Dim pInfo As PropertyInfo = GetType(PivotGridHeadersAreaHitInfo).GetProperty("HeadersViewInfo", BindingFlags.Instance Or BindingFlags.NonPublic)
                Dim headersViewInfo As PivotHeadersViewInfoBase = TryCast(pInfo.GetValue(hitInfo.HeadersAreaInfo, Nothing), PivotHeadersViewInfoBase)
                Dim headerViewInfo As PivotHeaderViewInfo = TryCast(headersViewInfo(GetFieldItem(pivot, hitInfo.HeaderField)), PivotHeaderViewInfo)
                Dim mInfo As MethodInfo = GetType(PivotHeaderViewInfo).GetMethod("ShowSummariesMenu", BindingFlags.NonPublic Or BindingFlags.Instance)
                mInfo.Invoke(headerViewInfo, New Object() {})
            End If
        End Sub

        Private Function GetFieldItem(ByVal pivot As PivotGridControl, ByVal field As PivotGridField) As DevExpress.XtraPivotGrid.Data.PivotFieldItemBase
            Return CType(pivot, IPivotGridViewInfoDataOwner).DataViewInfo.GetFieldItem(field)
        End Function
    End Class
End Namespace
